test/tests/functional/pbs_cgroups_hook.py

```diff
@@ -0,0 +1,1120 @@
+        jid = self.server.submit(j)
+        a = {'job_state': 'R'}
+        self.server.expect(JOB, a, jid)
+        self.server.status(JOB, [ATTR_o, ATTR_e], jid)
+        (o, e) = (j.attributes[ATTR_o], j.attributes[ATTR_e])
+        time.sleep(1)
+        rv = True
+        try:
+            self.mom.log_match("%s is not in " % (self.mom.shortname) +
+                               "the approved host list: " +
+                               "\['NonexistentNode'\]",
+                               regexp=True,
+                               max_attempts=5,
+                               starttime=self.server.ctime)
+        except:
+            rv = False
+        self.wait_and_remove(o.split(':')[1])
+        self.assertTrue(rv)
+
+    def test_cgroup_qstat_resources(self):
+        """
+        Test to verify that cgroups are reporting usage for
+        mem, and vmem in qstat
+        """
+        name = 'CGROUP14'
+        self.load_config(self.cfg3)
+        a = {'Resource_List.select': '1:ncpus=1:mem=500mb',
+             ATTR_N: name}
+        j = Job(TEST_USER, attrs=a)
+        j.create_script(self.eatmem_job3)
+        jid = self.server.submit(j)
+        a = {'job_state': 'R'}
+        self.server.expect(JOB, a, jid)
+        self.server.status(JOB, [ATTR_o, ATTR_e], jid)
+        (o, e) = (j.attributes[ATTR_o], j.attributes[ATTR_e])
+        self.logger.info("OUTPUT: %s" % o)
+        resc_list = ['resources_used.cput']
+        resc_list += ['resources_used.mem']
+        resc_list += ['resources_used.vmem']
+        qstat1 = self.server.status(JOB, resc_list, id=jid)
+        for q in qstat1:
+            self.logger.info("Q1: %s" % q)
+        cput1 = qstat1[0]['resources_used.cput']
+        mem1 = qstat1[0]['resources_used.mem']
+        vmem1 = qstat1[0]['resources_used.vmem']
+        time.sleep(10)
+        qstat2 = self.server.status(JOB, resc_list, id=jid)
+        for q in qstat2:
+            self.logger.info("Q2: %s" % q)
+        cput2 = qstat2[0]['resources_used.cput']
+        mem2 = qstat2[0]['resources_used.mem']
+        vmem2 = qstat2[0]['resources_used.vmem']
+        self.wait_and_remove(o.split(':')[1])
+        self.assertNotEqual(cput1, cput2)
+        self.assertNotEqual(mem1, mem2)
+        self.assertNotEqual(vmem1, vmem2)
+
+    def test_cgroup_reserve_mem(self):
+        """
+        Test to verify that the mom reserve memory for OS
+        when there is a reserve mem request in the config
+        """
+        self.load_config(self.cfg3)
+        self.momA.stop()
+        self.server.expect(NODE, {'state': 'down'},
+                           self.momA.shortname, interval=3)
+        self.momA.delete_vnode_defs()
+        self.momA.start()
+        self.server.expect(NODE, {'state': 'free'},
+                           self.momA.shortname, interval=3)
+        vmem = self.server.status(NODE, 'resources_available.vmem')
+        self.logger.info("%s" % str(vmem))
```

Comment:
modify the message to say that this is vmem value

Reply:
Done.